### PR TITLE
Setup Coroutine: Update necessary dependencies.

### PR DESCRIPTION
**Pull request by Wadeewee**
https://github.com/nimblehq/android-templates/projects/3#card-55575279

## What happened 🤔
To continue to have a new template for Coroutine, we must initial setup all of about necessary dependencies before work on this template.

## Insight 👀

- Update CI workflow for support both of RxJavaTemplate and CoroutineTemplate.
- Update README.md.
- Update build.gradle with necessary dependencies.
- Add basic structure.

## PoW (a.k.a Proof Of Work)💪

Screen Shot 2564-03-11 at 12 33 46

Screen Shot 2564-03-11 at 12 33 46

Screen Shot 2564-03-11 at 12 35 08

Screen Shot 2564-03-11 at 12 35 25

### CODE REVIEW

File: CoroutineTemplate/app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<resources>
+    <string name="app_name">Nimble Template</string>
+
+    <string name="coroutine_template_label">Coroutine Template</string>
```
[the remark this answers is not in the document]
Comment (Tuubz) [reply CoroutineTemplate/app/src/main/res/values/strings.xml:5]: Might need to rename for `RxJavaTemplate` as well then 🙏

**Review by Tuubz — CHANGES_REQUESTED** [review]
Rest LGTM 🙏

**Review by Tuubz — APPROVED** [review]

**Review by Tuubz — APPROVED** [review]
Oh, seems like I already approved it before 🙈

File: CoroutineTemplate/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Android Templates: Kotlin
```
Comment (luongvo) [inline CoroutineTemplate/README.md:1]: This template is not focused on Kotlin, should it be Coroutine?

File: CoroutineTemplate/app/src/main/java/co/nimblehq/coroutine/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package co.nimblehq.coroutine
+
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+
+class MainActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
```
Comment (luongvo) [inline CoroutineTemplate/app/src/main/java/co/nimblehq/coroutine/MainActivity.kt:7]: I think we are aligning together to keep 1 breakline on top of class 💭

File: CoroutineTemplate/build.gradle [hunk 4]
```diff
@@ -0,0 +1,77 @@
+// Top-level build file where you can add configuration options common to all sub-projects/modules.
+
+buildscript {
+    ext {
+        build_gradle_version = '4.1.1'
+
+        android_version_code = 9
+        android_version_name = "0.7.0"
```
Comment (luongvo) [inline CoroutineTemplate/build.gradle:8]: which version should we start?

File: CoroutineTemplate/build.gradle [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        android_support_version = '1.1.0'
+        constraint_layout_version = '2.0.0-alpha3'
+        androidx_core_ktx_version = '1.3.0'
+
+        kotlin_version = '1.4.30'
+        kotlinx_coroutines_version = '1.4.1'
+
+        android_test_junit_version = '1.1.1'
+        test_espresso_version = "3.1.0"
+        test_junit_version = '4.13'
```
Comment (luongvo) [inline CoroutineTemplate/build.gradle:22]: please keep them in a correct order, test components should start with test prefix

File: CoroutineTemplate/build.gradle [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+task clean(type: Delete) {
+    delete rootProject.buildDir
+}
+
+detekt {
+    toolVersion = "1.14.0"
+
+    input = files("app/src/main/java", "app/src/main/kotlin", "data/src/main/kotlin", "data/src/main/java")
```
Comment (luongvo) [inline CoroutineTemplate/build.gradle:55]: I think we only have `java` source folder

File: CoroutineTemplate/build.gradle (hunk 4 above)
Comment (Wadeewee) [reply CoroutineTemplate/build.gradle:8]: Oh, I think we can start with 0.1.0 🙏  What do you think? @Tuubz @chornerman

File: CoroutineTemplate/README.md (hunk 2 above)
Comment (Tuubz) [reply CoroutineTemplate/README.md:1]: Might need to change this in `RxJavaTemplate` as well 🙏

File: CoroutineTemplate/build.gradle (hunk 6 above)
Comment (Wadeewee) [reply CoroutineTemplate/build.gradle:55]: @luongvo Should we remove `"app/src/main/kotlin"` in JavaTemplate as well 🤔 cc @Tuubz

Comment (Tuubz) [reply CoroutineTemplate/build.gradle:55]: I agree

- For `CoroutineTemplate` we can keep it as: `"app/src/main/java"`
- For `RxJavaTemplate` we can keep it as: `"app/src/main/java", "data/src/main/java"`

What do you think?

Comment (luongvo) [reply CoroutineTemplate/build.gradle:55]: Just re-checked, no kotlin source folder is being used, you can remove it 👌 

2 places
image

**Comment by Wadeewee** [comment]
Updated 🚀

**Review by luongvo — APPROVED** [review]

**Review by hoangnguyen92dn — APPROVED** [review]
LGTM

File: CoroutineTemplate/app/src/main/res/values/colors_pallete.xml [hunk 7]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <!-- Declaring colors with their real name from the style guide we get -->
+    <color name="light_blue">#3F51B5</color>
+    <color name="dark_blue">#303F9F</color>
+    <color name="pink">#FF4081</color>
```
Comment (lydiasama) [inline CoroutineTemplate/app/src/main/res/values/colors_pallete.xml:6]: Can we set all of them with the real color name from https://www.htmlcsscolor.com/hex/3F51B5 ?

**Review by lydiasama — COMMENTED** [review]
The rest LGTM.

File: CoroutineTemplate/app/src/main/res/values/colors_pallete.xml (hunk 7 above)
Comment (Tuubz) [reply CoroutineTemplate/app/src/main/res/values/colors_pallete.xml]: Updated 🤖

**Comment by Tuubz** [comment]
Done ✅ 

Please merge 🙏

File: CoroutineTemplate/app/src/main/res/values/colors_pallete.xml [hunk 8]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <!-- Declaring colors with their real name from the style guide we get -->
+    <color name="free_speech_blue">#3F51B5</color>
+    <color name="tory_blue">#303F9F</color>
+    <color name="violet_red">#FF4081</color>
```
Comment (luongvo) [inline CoroutineTemplate/app/src/main/res/values/colors_pallete.xml:6]: yes, but I think we should put the main color at first, e.g. red_violet, red_rose 😂

Comment (Tuubz) [reply CoroutineTemplate/app/src/main/res/values/colors_pallete.xml:6]: omg 🙈

Comment (Tuubz) [reply CoroutineTemplate/app/src/main/res/values/colors_pallete.xml]: Done ✅ ✅ ✅ ✅

**Review by luongvo — APPROVED** [review]

**Review by lydiasama — APPROVED** [review]

**Pull request merged**